internal/domain/filter.go

```diff
@@ -686,6 +702,39 @@ func containsMatch(tags []string, filters []string) bool {
 	return false
 }
 
+func containsAllMatch(tags []string, filters []string) bool {
+	for _, filter := range filters {
+		if filter == "" {
+			continue
+		}
+		filter = strings.ToLower(filter)
+		filter = strings.Trim(filter, " ")
+		found := false
+
+		for _, tag := range tags {
+			if tag == "" {
+				continue
+			}
+			tag = strings.ToLower(tag)
+
+			a := strings.ContainsAny(filter, "?|*")
```

Comment:
You could reorganize the checking logic. First check whether tag == filter and then wrap the alternative logic in an else statement and then check for "a".